### PR TITLE
INT-4049: FileSplitter: JSON File Markers

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-4049

### CODE REVIEW

**Comment by garyrussell** [comment]
Note: `FileSplitter` changed to return message builders for markers in 4.3 - now a supported return type.

File: spring-integration-file/src/main/resources/org/springframework/integration/file/config/spring-integration-file-4.3.xsd [hunk 1]
```diff
@@ -717,8 +717,22 @@ Only files matching this regular expression will be picked up by this adapter.
...
+				<xsd:simpleType>
+					<xsd:union memberTypes="xsd:boolean xsd:string"/>
+				</xsd:simpleType>
+			</xsd:attribute>
+			<xsd:attribute name="markers-json" use="optional" default="false">
+				<xsd:annotation>
+					<xsd:documentation>
+						When 'markers' is true, if this is 'true', the message payload of the marker
+						is a JSON String representation of the marker object.
+						Requires a supported JSON implementation on the class path.
```
Comment (artembilan) [inline spring-integration-file/src/main/resources/org/springframework/integration/file/config/spring-integration-file-4.3.xsd]: I think "JSON processor library" would sound better.

File: src/reference/asciidoc/file.adoc [hunk 2]
```diff
@@ -818,33 +819,38 @@ Other payload types will be emitted unchanged.
...
 They enable the downstream processing to know when a file has been completely processed.
+In addition, a header `file_marker` containing `START` or `END` are added to these messages.
 The 'END' marker includes a line count.
 Default: `false`.
 When `true`, `apply-sequence` is `false` by default.
+Also see `markers-json`.
 
-<4> Set to `false` to disable the inclusion of `sequenceSize` and `sequenceNumber` headers in messages.
+<4> When `markers` is true, set this to `true` and the `FileMarker` objects will be converted to a JSON String.
+Requires a supported JSON implementation on the classpath (Jackson, Boon).
```
Comment (artembilan) [inline src/reference/asciidoc/file.adoc]: DITO.

File: src/reference/asciidoc/file.adoc [hunk 3]
```diff
@@ -866,3 +872,12 @@ using the `stream` option to retrieve a file, starting with _version 4.3_, the s
...
 information about these facilities.
+
+When using Java configuration, an additional constructor is available:
+
+[source, java]
+----
+public FileSplitter(boolean iterator, boolean markers, boolean markersJson)
+----
+
+When `markersJson` is true, the markers will be represented as a JSON string, as long as a suitable JSON implementation such as Jackson or Boon is on the classpath.
```
Comment (artembilan) [inline src/reference/asciidoc/file.adoc]: DITO.
Plus will add `What's New` note and modify `FtpServerOutboundTests-context.xml` to chech that new header work well:
```
<int:service-activator input-channel="markers"
	  expression="headers['file_marker']?.equals('END') ? headers['file_remoteSession'].close() : null"/>
```

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/d521031a7d370c67125403e49932fb9bdffe8c26

**Pull request closed without merging**